### ISSUE TITLE
Bug? In some situations, unknown registers do not make witgen fail

### ISSUE THREAD

**Issue by chriseth** [powdr-labs/powdr#253]
If we run `cargo run -r asm ./test_data/asm/simple_sum.asm -o /tmp/k5   -i 300,1`, then witness generation does not fail.

It claims that the sum of one non-existing element is 300. At step 9, witgen outputs the following:
```
Input query: Index 0 -> 300
===== Row 9:
    Assembly.pc = 7
    Assembly.reg_write_X_A = 1
    Assembly.X_read_free = -1
    Assembly.read_X_A = 1
    Assembly.X_free_value = 300
    Assembly.reg_write_X_CNT = 0
    Assembly.CNT = 0
    Assembly.instr_jmpz = 0
    Assembly.instr_jmpz_param_l = 0
    Assembly.instr_jmp = 0
    Assembly.instr_jmp_param_l = 0
    Assembly.instr_dec_CNT = 0
    Assembly.instr_assert_zero = 0
    Assembly.X_const = 0
    Assembly.read_X_CNT = 0
    Assembly.read_X_pc = 0
    Assembly.X = <unknown>
    Assembly.A = <unknown>
    Assembly.XInv = <unknown>
    Assembly.XIsZero = <unknown>
```

And then it continues with `X = 0` and `A = 0`.

My assumption is that this is because there is a (even unique) value to read from the prover (even if the prover does not provide it) so that the sum matches.

Is this a problem? After all, prover inputs are nondeterministic and we do not check them.
`

**Comment by chriseth** [comment]
Ok, this may be fine, but if you run `sum.rs` without any prover input, at some point `pc` gets `<unknown>` and the witness generation still continues...

**Comment by Schaeff** [comment]
I think I had something related before [this commit](https://github.com/chriseth/powdr/pull/151/commits/c7864aa0880618ee90dbeb30d30dbd8392fcc316#diff-7f4aad66400fe52aa7613df96ab4235523b6c30f98dac442d4a2653036345a34L33-L43): the prover query returned `None` and somehow witgen still worked, but the witness was invalid.

**Comment by leonardoalt** [comment]
So is this a bug?

**Comment by chriseth** [comment]
I think this should be fixed with #324 now, let me check.

**Comment by leonardoalt** [comment]
The current output now says:

```
Deducing witness columns...
Not enough inputs provided! Index 2 out of bounds
Not enough inputs provided! Index 2 out of bounds
```

Closing then.
